app/modules/uart.c

```diff
@@ -8,19 +8,19 @@
 #include "c_string.h"
 #include "rom.h"
 
-static lua_State *gL = NULL;
 static int uart_receive_rf = LUA_NOREF;
 bool run_input = true;
 bool uart_on_data_cb(const char *buf, size_t len){
+  lua_State *L = lua_getstate();
   if(!buf || len==0)
     return false;
   if(uart_receive_rf == LUA_NOREF)
     return false;
-  if(!gL)
+  if(!L)
```

Comment:
Yup, I'd spotted this one myself.  The tl;dr is no.   The lua_State used by `lua_getstate()` is set up pretty much 1st thing in `lua_main()` so the only time it can be zero is if `lua_main()` hasn't been called.  Any library API call can only be executed after `lua_main()` has been run, so the only potential issue is with a driver ISR triggered task if the ISR is set up by the `platform_init()`.  The one case here is the UART driver and we've already fixed this race condition (#1142).